packages/app/ui/icon/Icon.tsx

```diff
@@ -0,0 +1,69 @@
+import Image from "next/image";
+import PlusIcon from "../../public/assets/icons/plus.svg";
+import HamburgerIcon from "../../public/assets/icons/hamburger.svg";
+import CloseIcon from "../../public/assets/icons/close.svg";
+import BlocksIcon from "../../public/assets/icons/blocks.svg";
+import CaretLeftIcon from "../../public/assets/icons/caret-left.svg";
+import CaretDownIcon from "../../public/assets/icons/caret-down.svg";
+import CaretRightIcon from "../../public/assets/icons/caret-right.svg";
+import SwapIcon from "../../public/assets/icons/swap.svg";
+import WarningIcon from "../../public/assets/icons/warning.svg";
+import CheckIcon from "../../public/assets/icons/check.svg";
+import ArrowLeftIcon from "../../public/assets/icons/arrow-left.svg";
+import ArrowExternalIcon from "../../public/assets/icons/arrow-external.svg";
+
+export type IconName =
+  | "close"
+  | "plus"
+  | "menu"
+  | "blocks"
+  | "check"
+  | "arrow-left"
+  | "arrow-external"
+  | "warning"
+  | "swap"
+  | "caret-down"
+  | "caret-left"
+  | "caret-right";
+
+export const iconMap: Record<IconName, any> = {
+  blocks: BlocksIcon,
+  plus: PlusIcon,
+  close: CloseIcon,
+  menu: HamburgerIcon,
+  check: CheckIcon,
+  "arrow-left": ArrowLeftIcon,
+  "arrow-external": ArrowExternalIcon,
+  warning: WarningIcon,
+  swap: SwapIcon,
+  "caret-down": CaretDownIcon,
+  "caret-left": CaretLeftIcon,
+  "caret-right": CaretRightIcon,
+};
+
+const defaultIcon = iconMap["blocks"];
+
+const getIcon = (name: IconName) => iconMap[name] || defaultIcon;
```

Comment:
I think the app should break if there isn't the correct Icon. Or at least render a dummy element. But I think throwing an error is better than a dummy element, gives no room for errors.

Reply:
Maybe we can remove this,  it's typesafe so it has linting error and will break on compiling.

Reply:
addressed here: 074aec04d6e31c699a13ec484e0df09d978537f4